sdk/python/feast/feature_view.py

```diff
@@ -379,3 +388,23 @@ def infer_features_from_batch_source(self, config: RepoConfig):
                     "FeatureView",
                     f"Could not infer Features for the FeatureView named {self.name}.",
                 )
+
+    def set_projection(self, feature_view_projection: FeatureViewProjection) -> None:
+        """
+        Setter for the projection object held by this FeatureView. Performs checks to ensure
+        the projection is consistent with this FeatureView before doing the set.
+
+        Args:
+            feature_view_projection: The FeatureViewProjection object to set this FeatureView's
+                'projection' field to.
+        """
+        assert feature_view_projection.name == self.name
```

Comment:
We should not be using `assert` since it can be disabled inadvertently through optimization flags.